resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -368,6 +369,19 @@ class YarnClusterSuite extends BaseYarnClusterSuite {
     )
     checkResult(finalState, result, "true")
   }
+
+  test("SPARK-34472: ivySettings file should be localized on driver in cluster mode") {
+
+    val emptyIvySettings = File.createTempFile("ivy", ".xml")
+    FileUtils.write(emptyIvySettings, "<ivysettings />", StandardCharsets.UTF_8)
+
```

Comment:
Can we use NIO for these? No need for commons-io now that NIO supports this kind of stuff built-in.

```
val emptyIvySettings = Files.createTempFile(...)
Files.write(emptyIvySettings, Seq(...))
```

Reply:
I changed this to use Guava's Files which is used in many places within this file. Can I create a followup PR to replace these with NIO?

Reply:
Just saw this comment, sounds fine to me.